pkg/paravirt/paravirt_protected_entity_type_manager.go

```diff
@@ -166,3 +186,230 @@ func (this *ParaVirtProtectedEntityTypeManager) getDataTransports(id astrolabe.P
+	}
+	this.logger.Info("CreateFromMetadata: decoded snapshotID: %s", snapIDDecoded)
+
+	apiGroup := ""
+	kind := "PersistentVolumeClaim"
+
+	// Decode metadata, change namespace of PVC
+	// from Guest Cluster namespace to Supervisor Cluster namespace,
+	// and encode again before calling CreateFromMetadata
+	// NOTE(xyang): We need to make assumption that StorageClass is already created
+	// in Supervisor and Synchronized to Guest Cluster
+	// StorageClass in Guest must have StorageClass name of Supervisor
+	// parameters:
+	//   svStorageClass: gc-storage-profile
+	svcPVC := v1.PersistentVolumeClaim{}
+	err = svcPVC.Unmarshal(metadata)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to retrieve PVC")
+	}
+	svcPVC.Namespace = this.svcNamespace
+	this.logger.Infof("StorageClassName is %s in Supervisor PVC: %s/%s", *svcPVC.Spec.StorageClassName, svcPVC.Name, svcPVC.Namespace)
+	svcPVCData, err := svcPVC.Marshal()
+	if err != nil {
+		return nil, errors.Wrapf(err, "Could not marshal SVC PVC data for %v",
+			svcPVC.Name, svcPVC.Namespace)
+	}
+
+	// We need to construct the snapshotID in CloneFromSnapshot for Supervisor in this format:
+	// pvc:test-gc-e2e-demo-ns/test-pvc:aXZkOjFlNDZiYjRkLWIzZjAtNDBkNS05Y2E4LTNiYWU2ZjU5NTk1NTpkMmUyMjcxNi05YTYxLTQ2OGEtYjBhYy0xOTJkMmJkMWI3ZjU
+	// aXZkOjFlNDZiYjRkLWIzZjAtNDBkNS05Y2E4LTNiYWU2ZjU5NTk1NTpkMmUyMjcxNi05YTYxLTQ2OGEtYjBhYy0xOTJkMmJkMWI3ZjU will be decoded to ivd:1e46bb4d-b3f0-40d5-9ca8-3bae6f595955:ea4e347a-be29-4e5b-a626-725b83f168fcbase64 by PVC PETM
+	newSnapshotID := "pvc:" + svcPVC.Namespace + "/" + svcPVC.Name + ":" + snapIDDecoded
+	this.logger.Infof("CreateFromMetadata: constructed new SnapshotID: %s", newSnapshotID)
+
+	// Build a CloneFromSnapshot API object for Supervisor Cluster
+	// and wait for its phase to become Completed, Failed, or Canceled.
+	// metadata should be []byte form of PVC so supervisor cluster
+	// PVC PETM will call CreateFromMeta which will create a PVC
+	// in the Supervisor Cluster
+	this.logger.Info("Creating a CloneFromSnapshot CR")
+	svcCloneFromSnapshot, err := snapshotUtils.CloneFromSnapshopRef(ctx, this.svcBackupDriverClient, newSnapshotID, svcPVCData, &apiGroup, kind, this.svcNamespace, *backupRepo, []backupdriverv1.ClonePhase{backupdriverv1.ClonePhaseCompleted, backupdriverv1.ClonePhaseFailed, backupdriverv1.ClonePhaseCanceled}, this.logger)
+	this.logger.Infof("CreateFromMetadata: finished waiting for CloneFromSnapshot's status to be completed, failed, or canceled in the Supervisor Cluster")
+
+	if err != nil {
+		this.logger.Errorf("Failed to create a cloneFromSnapshot CR: %v", err)
+		return nil, err
+	}
+
+	// Get CloneFromSnapshot from Supervisor
+	svcClone, err := this.svcBackupDriverClient.CloneFromSnapshots(this.svcNamespace).Get(context.TODO(), svcCloneFromSnapshot.Name, metav1.GetOptions{})
+	if err != nil {
+		this.logger.Errorf("Failed to get the cloneFromSnapshot CR %s/%s: %v", this.svcNamespace, svcCloneFromSnapshot.Name, err)
```

Comment:
This is repeated several times, construct the string and use that string variable.